File: src/baseDatos/Serializador.java
package baseDatos;

import java.io.File;
import java.io.FileNotFoundException;
import java.io.IOException;
import java.io.FileOutputStream;
import java.io.ObjectOutputStream;
import java.io.PrintWriter;
import gestorAplicacion.Registro;

public class Serializador {
    private static File temp = new File("src\\baseDatos\\temp");
    
    public static void serializador(){
        FileOutputStream fos;
        ObjectOutputStream oos;
        File[] docs = temp.listFiles();
        PrintWriter pw;
        
        for(File f : docs){
            
            try{
                pw = new PrintWriter(f);
            }
            catch (FileNotFoundException e){
                e.printStackTrace();
            }      
        }
        for(File f : docs){
            
            if(f.getAbsolutePath().contains("profesores")){
                try{
                    fos = new FileOutputStream(f);
                    oos = new ObjectOutputStream(fos);
                    oos.writeObject(Registro.getProfesores());
                }
                catch (FileNotFoundException e){
                    e.printStackTrace();
                }
                catch (IOException e){
                    e.printStackTrace();
                }
            }
            else if(f.getAbsolutePath().contains("estudiantes")){
                try{
                    fos = new FileOutputStream(f);
                    oos = new ObjectOutputStream(fos);
                    oos.writeObject(Registro.getEstudiantes());
                }
                catch (FileNotFoundException e){
                    e.printStackTrace();
                }
                catch (IOException e){
                    e.printStackTrace();
                }
            }
            else if(f.getAbsolutePath().contains("admins")){
                try{
                    fos = new FileOutputStream(f);
                    oos = new ObjectOutputStream(fos);
                    oos.writeObject(Registro.getAdmins());
                }
                catch (FileNotFoundException e){
                    e.printStackTrace();
                }
                catch (IOException e){
                    e.printStackTrace();
                }
            }
            else if(f.getAbsolutePath().contains("cursos")){
                try{
                    fos = new FileOutputStream(f);
                    oos = new ObjectOutputStream(fos);
                    oos.writeObject(Registro.getCursos());
                }
                catch (FileNotFoundException e){
                    e.printStackTrace();
                }
                catch (IOException e){
                    e.printStackTrace();
                }
            }
            else if(f.getAbsolutePath().contains("estudiantesMatriculados")){
                try{
                    fos = new FileOutputStream(f);
                    oos = new ObjectOutputStream(fos);
                    oos.writeObject(Registro.getEstudiantesMatriculados());
                }
                catch (FileNotFoundException e){
                    e.printStackTrace();
                }
                catch (IOException e){
                    e.printStackTrace();
                }
            }
        }
    }
}

File: PYTHON/src/baseDatos/Serializador.py
import pickle
from gestorAplicacion.clasesDeUsuario.Registro import Registro

class Serializador:
    @classmethod
    def serializador(cls):
        file = open("baseDatos/temp/profesores.pkl", "wb")
        pcs = Registro.getProfesores()
        pickle.dump(pcs, file)
        file.close()

        file = open("baseDatos/temp/estudiantes.pkl", "wb")
        pcs = Registro.getEstudiantes()
        pickle.dump(pcs, file)
        file.close()

        file = open("baseDatos/temp/admins.pkl", "wb")
        pcs = Registro.getAdmins()
        pickle.dump(pcs, file)
        file.close()

        file = open("baseDatos/temp/estudiantesMatriculados.pkl", "wb")
        pcs = Registro.getEstudiantesMatriculados()
        pickle.dump(pcs, file)
        file.close()

        file = open("baseDatos/temp/estimulosProfesores.pkl", "wb")
        pcs = Registro.getEstimulosProfesores()
        pickle.dump(pcs, file)
        file.close()

        file = open("baseDatos/temp/estimulosEstudiantes.pkl", "wb")
        pcs = Registro.getEstimulosEstudiantes()
        pickle.dump(pcs, file)
        file.close()

        file = open("baseDatos/temp/cursos.pkl", "wb")
        pcs = Registro.getCursos()
        pickle.dump(pcs, file)
        file.close()